src/main/java/org/jitsi/videobridge/SctpConnection.java

```diff
@@ -1118,9 +1121,6 @@ public void run()
                     }
                 }
 
-                if (sctpSocket == null)
-                    break;
-
```

Comment:
I get the reason why the same check is necessary before the receive (after reading your PR comment). But I don't get why it's no longer necessary after the receive. Generally, a receive usually takes significantly long time to return so checks from before may not hold afterwards. In the case that may or may not be true, I'm just saying that it's not obvious to me.

On the other hand, the for loop bellow will repeatedly access sctpSocket so it's not like it will not become null inside the loop (I guess).

Anyway, I suppose I mostly wanted to express my feeling that something's amiss around these null checks outside a loop and accesses to a field in a loop.

It's outside the scope of this PR, of course, but I really feel that this method should be broken down into multiple smaller ones in order to aid reading and understanding.